### DISCUSSION TITLE
Trade confirmations not working / importing SDA files not working

### DISCUSSION THREAD

**Discussion by mirtisdyleris** [dyc3/steamguard-cli#127] [Q&A] [answered]
Hello.

My first problem was that I couldnt just Import my files from SDA. It doesnt just do it. If I put in the data from the old file into a .maFile of an account I natively setup in steamguard-cli it works. The difference seems to be the order in which the individual things in the file are set up and maybe something missing in the SDA files.

After I imported an account the manual way I tried to send out a trade with this and didnt work. I put in steamguard trade and then after pressing a random button (as it tells me to do) I put in a and then enter. Didnt work, it also said that there is nothing to confirm in the stuff that came up when I put in the command. Even if I do steamguard trade -a it says it accepts everything but it doesnt.

Last question, will steamguard trade -a also accept steam market confirmations? :)

**Comment by dyc3** [comment, marked as answer]
> My first problem was that I couldnt just Import my files from SDA.

What exactly did you try? Did you use `steamguard import`? If you could provide a censored maFile to test with, that would be the most helpful.

> Didnt work, it also said that there is nothing to confirm in the stuff that came up when I put in the command.

If you could provide debug or trace outputs, that would be super helpful.
Debug: `-v`
Trace: `-vv` (Warning: prints sensitive information, you will need to censor it manually)

> will `steamguard trade -a` also accept steam market confirmations?

Yeah, it should accept all types of confirmations.

**Reply by mirtisdyleris** [reply]
> What exactly did you try? Did you use `steamguard import`? If you could provide a censored maFile to test with, that would be the most helpful.

I didnt use steamguard import, because I seem to either not get the syntax right or it simply doesnt work. Here is a censored maFile: https://pastebin.com/nksnctTF


> If you could provide debug or trace outputs, that would be super helpful.
> Debug: `-v`
> Trace: `-vv` (Warning: prints sensitive information, you will need to censor it manually)

Maybe Im stupid, but steamguard trade -v or -vv doesnt work for me. steamguard -v worked, but the issue arent the auth codes, they work fine, just the confirmations dont work. 
"steamguard trade -vv (both with -v and -vv!)
error: Found argument '-v' which wasn't expected, or isn't valid in this context"

**Reply by dyc3** [reply]
> I didnt use steamguard import, because I seem to either not get the syntax right

`steamguard-cli import --help`

It should just be `steamguard import path/to/file`.

> steamguard trade -v or -vv

Try `steamguard -v trade`, should've been more specific.

**Reply by mirtisdyleris** [reply]
> Try `steamguard -v trade`, should've been more specific.

https://pastebin.com/xrhDTNMH



> It should just be `steamguard import path/to/file`.

path to which file?

If i put in the path to a maFile this error comes:

INFO - reading manifest from /home/oem/.config/steamguard-cli/maFiles
ERROR - Failed to import account: /media/oem/xxx/Program Files (x86)/Steam Desktop Authenticator/maFiles/xxx.maFile invalid type: null, expected a string at line 1 column 577
INFO - Saving manifest and accounts...

If I put in the manifest this comes:

INFO - reading manifest from /home/oem/.config/steamguard-cli/maFiles
ERROR - Failed to import account: /media/oem/xxx/Program Files (x86)/Steam Desktop Authenticator/maFiles/manifest.json missing field `account_name` at line 1 column 4690
INFO - Saving manifest and accounts...

Thats what it showed before also and thats why I assumed I got the syntax or smth different wrong.

**Reply by dyc3** [reply]
Looking at that log, steam correctly returned that there are no confirmations for the account. IIRC, if you are buying items on the steam market then you don't need to confirm it, only market sells.

And yeah, you are supposed to import the maFile. Looks like the parser is failing.

In the maFile you are trying to import, `OAuthToken` is set to null, which is a required field. I'll open an issue to fix that. (#128) As a workaround, try setting the `Session` field to `null`.

**Reply by mirtisdyleris** [reply]
> Looking at that log, steam correctly returned that there are no confirmations for the account. IIRC, if you are buying items on the steam market then you don't need to confirm it, only market sells.

I know. Issue being that I sent out a trade and when I open SDA on Windows with that maFiles the trades show up and I was able to confirm them. So wtf is going on? :D

**Reply by mirtisdyleris** [reply]
> In the maFile you are trying to import, `OAuthToken` is set to null, which is a required field. I'll open an issue to fix that. As a workaround, try setting the `Session` field to `null`.

Sadly that didnt work either:

INFO - reading manifest from /home/oem/.config/steamguard-cli/maFiles
ERROR - Could not load accounts: No such file or directory (os error 2)

both with ""Session":{"SessionID":"null"," and with ""Session":{"null","

**Reply by dyc3** [reply]
remove the quotes from around `null`. `"null"` -> `null`.
Should look like
```json
{
    ...
    "Session": null
}
```

**Reply by dyc3** [reply]
> Could not load accounts: No such file or directory

Also, check your manifest, see if there are files that don't exist. I didn't quite realize it would prevent `steamguard` from running, I'll open an issue for that. (#129)

**Reply by mirtisdyleris** [reply]
> > Looking at that log, steam correctly returned that there are no confirmations for the account. IIRC, if you are buying items on the steam market then you don't need to confirm it, only market sells.
> 
> I know. Issue being that I sent out a trade and when I open SDA on Windows with that maFiles the trades show up and I was able to confirm them. So wtf is going on? :D


Im going to try what u said, but did u see that?

**Reply by dyc3** [reply]
It could be requesting confirmations for the wrong account. Do you have multiple accounts? Otherwise, it might as well be attributed to steam being weird. Can't parse the confirmations if steam doesn't give return them.

**Reply by mirtisdyleris** [reply]
> It could be requesting confirmations for the wrong account. Do you have multiple accounts? Otherwise, it might as well be attributed to steam being weird. Can't parse the confirmations if steam doesn't give return them.

I have multiple accounts, yes. But when SDA has no issues getting the confirmations, then it doesnt seem like a Steam problem? It also looks like it only checks a single account, but not all of em?

**Reply by dyc3** [reply]
Yes, by default, steamguard-cli only selects the first account to act on in the manifest. `steamguard --all ...` will select all of them.

**Reply by dyc3** [reply]
Please read `steamguard --help`

**Reply by mirtisdyleris** [reply]
> remove the quotes from around `null`. `"null"` -> `null`. Should look like
> 
> ```json
> {
>     ...
>     "Session": null
> }
> ```

Didnt work or I was too stupid to do it right. 

"Session": null,"SteamLogin":null, is now in it and it doesnt work.

**Reply by mirtisdyleris** [reply]
> Yes, by default, steamguard-cli only selects the first account to act on in the manifest. `steamguard --all ...` will select all of them.

Oh I see, thank you. So the only issue left is the parser not understanding the SDA files. Is there an option to autoaccept confirmations without needing to put in steamguard trade -a? Bc I wanna put 10s of thousands of items (cases) on the market and opening the terminal every 250 items would be kind of tedious.

**Reply by dyc3** [reply]
just loop it in bash

```bash
while true; do steamguard --all trade -a; sleep 20; done
```

**Reply by mirtisdyleris** [reply]
> Didnt work or I was too stupid to do it right.
> 
> "Session": null,"SteamLogin":null, is now in it and it doesnt work.

That sadly doesnt work, I guess for a quick fix I can just manually import the most important accounts instead of all 40 for now. If you have another workaround or if you fixed it please tell me about it.

**Reply by mirtisdyleris** [reply]
> just loop it in bash
> 
> ```shell
> while true; do steamguard --all trade -a; sleep 20; done
> ```

but thanks for that! :)

**Comment by 16546489** [comment]
Greetings.
I'm wondering how to use the import function the -h only gives me a brief information. 
I'm not a programmer but I been struggling with the setup for a while and I don't want to use windows because it an easier option.
So far I tried:
I tried copy pasting the mafile into the direct folder.
Tried using the import command and referring to the path where the mafile is in.
I always get this:
INFO reading manifest from /home/examplename/.config/steamgard-cli/maFlies
could not load accounts: could not find an entry in the manifest for this account. 
what I used for pathing was the exact same I tried putting it in as a string and tried it without it as well.
It would help if the -h gives me a little more detailed explanation considering casual linux users or an example with default options which is easy to copy.

**Reply by dyc3** [reply]
I'm confused about where you're having trouble, the help text (`steamguard-cli import -h`) specifically lists out the `--files` option

```
OPTIONS:
        --files <FILES>    Paths to one or more maFiles, eg. "./gaben.maFile"
```

So the full command should look like this:

```
steamguard import --files ./gaben.maFile
```
